### PR TITLE
Sherylueen appcomplianceautomation microsoft.app compliance automation 2024 06 27

### PR DESCRIPTION
**Pull request by Sherylueen**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by mentat9** [comment]
@Sherylueen - I removed the MergeRequested label: your PR requires ARM review first. It's in the PR queue - reviewers will get to it as soon as possible. Note, you should also refresh to fix the merge failure.

**Comment by mentat9** [comment]
@Sherylueen - To enable review, you will need to split this PR into two separate steps: first is to convert to typespec, then the second to move to the new API version. Otherwise, it's too hard for ARM reviewers to see what's changing.

**Comment by Sherylueen** [comment]
> @Sherylueen - To enable review, you will need to split this PR into two separate steps: first is to convert to typespec, then the second to move to the new API version. Otherwise, it's too hard for ARM reviewers to see what's changing.

Hi @mentat9 ~
Can I first move to the new API version, and then convert it to typespec?
And the spec and typespec change has already merged into PRSaaSMaster branch, Would this reduce the effort for ARM review？

**Comment by Sherylueen** [comment]
Hi @mentat9 ~
I have fixed the compile error caused by the 0.56.0 compiler, And all required checks are fixed.
Please help to review.
We rely on this PR to generate the CLIs, to meet our GA readiness requirements at 2024-06-27, so it's a little bit urgent.
Sorry for the inconvenience.

**Comment by mentat9** [comment]
> > @Sherylueen - To enable review, you will need to split this PR into two separate steps: first is to convert to typespec, then the second to move to the new API version. Otherwise, it's too hard for ARM reviewers to see what's changing.
> 
> Hi @mentat9 ~ Can I first move to the new API version, and then convert it to typespec? And the spec and typespec change has already merged into PRSaaSMaster branch, Would this reduce the effort for ARM review？

Yes, it's fine to do the API version change first in swagger, then move the new API version to typespec in a subsequent PR.
If these APIs were previously signed off by ARM, add the PR link in comments to help reviewers avoid re-reviewing.

**Comment by mentat9** [comment]
> Hi @mentat9 ~ I have fixed the compile error caused by the 0.56.0 compiler, And all required checks are fixed. Please help to review. We rely on this PR to generate the CLIs, to meet our GA readiness requirements at 2024-06-27, so it's a little bit urgent. Sorry for the inconvenience.

Also before merge, you will need to address the breaking change for Go. The Next Steps to Merge comment has the information you need.
Your PR is in the review queue for ARM review and will be picked up by this week's on-call ARM reviewer as soon as possible. Note, you can also attend daily office hours if you want extra help/attention.

**Comment by Sherylueen** [comment]
Hi @razvanbadea-msft ~
Here is the Merged PR in Private repo: https://github.com/Azure/azure-rest-api-specs-pr/pull/17908
The changes in this PR are reviewed by ARM team.
And I also pinged @Alancere to help me approve the Go sdk breaking change, Alancere helped me approved the same change in the previous PR.

**Comment by Sherylueen** [comment]
Hi @razvanbadea-msft ~
Here is the Merged PR in Private repo: https://github.com/Azure/azure-rest-api-specs-pr/pull/17908
The changes in this PR are reviewed by ARM team.
And I also pinged @Alancere to help me approve the Go sdk breaking change, Alancere helped me approved the same change in the previous PR.

**Comment by Sherylueen** [comment]
Hi @mentat9 and @razvanbadea-msft ~ Got feedback from @Alancere, he will help to approve go sdk breaking change, once the pr is passed ARM review. Thank you all~

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp
Comment (razvanbadea-msft) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp:26]: >Azure.ResourceManager.CommonTypes.Versions.v3 [](http://example.com/codeflow?start=23&length=45)

You should update the common types to version 5

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml
Comment (MaryGao) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml:20]: JS failed with no `service-dir` set in config and this would be used as prefix folder name in SDK repo and see example here https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml#L2C4-L2C15.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp
Comment (msyyc) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp:38]: Why is key not same with value like `v2024_06_27: "2024-06-27"`?

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml
Comment (msyyc) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml:20]: Python failed for lack of "service-dir". Here is example: https://github.com/Azure/azure-rest-api-specs/pull/29080/files#diff-eb92accee49d9c181cc391b6e60071d9b47754f4495fee29938194862ad2234b
image

Comment (qiaozha) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml:20]: @raych1 is it possible to provide tspconfig.yaml template and when they fill in their service name, those files can automatically replace the template args with real service name. I think we have met this kind of issue several times?

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp
Comment (Sherylueen) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/main.tsp]: Worry about this wrong version, this may caused we modified this version based on the v05-14 version.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml
Comment (Sherylueen) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml:26]: Hi @MaryGao~
I updated the tsconfig.yaml, the JS failed. Could you help me have a look~

**Comment by msyyc** [comment]
This PR is about migration from swagger to Typespec, so add SDK PM @josefree for awareness.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/SnapshotResource.tsp
Comment (msyyc) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/SnapshotResource.tsp:43]: Same with other interface, better to remove suffix `Resources`

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/ReportResource.tsp
Comment (msyyc) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/ReportResource.tsp:186]: I think the following breakings are caused by lack of @@clientName and @@OpenAPI.
image
Please check other places to make sure all 3 of @@clientName/@@ OpenAPI/@encodedName exist at the same time, otherwise it may cause unexpcected breakings in the future for some language SDK

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp
Comment (Sherylueen) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp]: Removed~

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/SnapshotResource.tsp
Comment (Sherylueen) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/SnapshotResource.tsp]: Removed the suffix Resources.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml
Comment (raych1) [reply specification/appcomplianceautomation/AppComplianceAutomation.Management/tspconfig.yaml:20]: @qiaozha , service team should reference the [TSP sample project](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md#service-folder-structure) to config SDK emitter in `tsp-config.yaml`

File: specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/stable/2024-06-27/appcomplianceautomation.json
Comment (msyyc) [inline specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/stable/2024-06-27/appcomplianceautomation.json:3058]: It seems this definition is not referred by any API.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp
Comment (msyyc) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp:82]: `@operationId` is for swagger and it is better to keep same with interface of typespec, in specific, please remove `s` for all `operationdId`, otherwise it will cause breaking changes of SDK when SDK generations changes from swagger to Typespec in the future.
nit: the suggestion applies for all operation.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/ReportResource.tsp
Comment (msyyc) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/ReportResource.tsp:137]:
```suggestion
  @operationId("Report_GetScopingQuestions")
```

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp
Comment (tadelesh) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp:24]: since this is the first ga, i suggest to use standard `ProxyResource` instead of keep full backward compatibility.

File: specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp
Comment (tadelesh) [inline specification/appcomplianceautomation/AppComplianceAutomation.Management/resources/EvidenceResource.tsp:62]:
```suggestion

```

suggest to remove this to follow arm standard for operation id.

File: specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/stable/2024-06-27/appcomplianceautomation.json
Comment (Sherylueen) [reply specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/stable/2024-06-27/appcomplianceautomation.json]: Fixed~

**Comment by Sherylueen** [comment]
Hi @madhura-np Could you help me to merge this PR?
This PR has been ARM reviewed and the breaking change of SDKs have also been approved.

**Review by rkmanda — APPROVED** [review]

**Pull request merged**
